search-app/src/cmr/search/services/query_execution/facets/hierarchical_v2_facets.clj

```diff
@@ -167,8 +172,12 @@
                          the part of the facets v2 response related to the passed in field."
   [recursive-parse-fn has-siblings-fn generate-links-fn field field-hierarchy elastic-aggregations]
   ;; Each value for this field has its own bucket in the elastic aggregations response
-  (for [bucket (get-in elastic-aggregations [field :buckets])
-        :let [value (:key bucket)
+  ;; Top level could be a field
+  (for [bucket (or (get-in elastic-aggregations [field :buckets] (get elastic-aggregations :buckets)))
+        :let [temporal? (some? (:key_as_string bucket))
+              value (if temporal?
```

Comment:
Is this something we are going to have to do for things other than temporal?  Is there going to be a condition for spatial or something else down the line,  or do we only expect to have this special instance of branching for temporal?

Reply:
I would guess, based on the initial use case that Chris has described and the general usefulness of facets for Earth data, that there will be more. 

I'm guessing that your implicit concern here is that a potentially significant bit of logic is special-cased and buried in a `for`? I'd definitely be +1 for moving this out ... just on general principal. That being said, I'd leave it up to Chris ... without an immediate need, it's a bit of personal preference.

Reply:
This is one of the ugly bits I mentioned. It is nested so many functions deep that this was the least intrusive way to make the change. I could move the code to do this check in a separate function and call it here, might be slightly less ugly.